src/peer_connection.cpp

```diff
@@ -5460,23 +5460,22 @@ namespace libtorrent
 	int peer_connection::wanted_transfer(int channel)
 	{
 		TORRENT_ASSERT(is_single_thread());
-		shared_ptr<torrent> t = m_torrent.lock();
 
-		const int tick_interval = (std::max)(1, m_settings.get_int(settings_pack::tick_interval));
+#if TORRENT_NO_FPU
+		int const tick_interval = (std::max)(1, m_settings.get_int(settings_pack::tick_interval) / 1000);
```

Comment:
This new expression is ill-conditioned from the numerical point of view and not equivalent to the previous ones. Also, the use of floating point arithmetic is only used in deprecated features in `master`.

Reply:
Well, then could you please explain me the meaning of both original expressions?

Reply:
other than the missing `* 2`, it all comes down to the fact that precision in integer division works better if the numerator have bigger module.

Reply:
I wouldn't be so sure. Consider the following input:
upload_rate = 1 000 000 000(1 GB/s)
tick_interval = 4500(4.5s)
Results would be:
Original expression: 9 000 000 000 = 9 GB
New expression: 4 000 000 000 = 4 GB
Precise result : 4.5 GB
In my opinion 4 is a way better approximation to 4.5 than 9 gigs.

Reply:
to be honest, I didn't pay attention to the correctness of the value returned in the method (you can be right about the dropping of the `*2`), only to the fact that you never do this kind of refactor in a numeric sensitive operation, you are definitely losing precision. Also, If I remember correctly, arvidn was moving away from float operations, which is a good idea IMO.

Reply:
I'm not sure why the rates are multiplied by 2, but I don't see any benefit to using floating point here. If you simply removed the `* 2` from the pure integer version it would yield exactly 4.5GB.

Reply:
That's exactly what it evaluates to if there is no FPU support.
```
int const tick_interval = (std::max)(1, m_settings.get_int(settings_pack::tick_interval) / 1000);
```
That is meant to be max of 1 second and floor of settings_pack::tick_interval in seconds.

Reply:
Ouch, it also appears that int(4.5GB) would give ~205MB. I think function's arithmetic should be covered with a test, to check it's accuracy for wide range of hypothetical input data.

Reply:
I think it's risky to use ``int`` operands here. It's actually worse than yielding ~205 MB, it yields undefined behaviour because of signed overflow.
I would prefer to avoid floating point arithmetic if it can be, and it looks like it's not necessary or particularly important in this case.